src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1344 @@ static ColliderCylinderInit D_80B37A5C = {
+    f32 lastFrame = Animation_GetLastFrame(&gWolfosBackflippingAnim);
+
+    Animation_Change(&this->skelAnime, &gWolfosBackflippingAnim, -1.0f, lastFrame, 0.0f, ANIMMODE_ONCE, -3.0f);
+    this->actionTimer = 0;
+    this->isInvisible = false;
+    this->action = WOLFOS_ACTION_TURN_TOWARDS_PLAYER;
+    this->actor.speedXZ = 6.5f;
+    this->actor.velocity.y = 15.0f;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_JUMP);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    EnWf_SetupAction(this, EnWf_TurnTowardsPlayer);
+}
+
+void EnWf_TurnTowardsPlayer(EnWf* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 1, 4000, 1);
+
+    if (this->actor.velocity.y >= 5.0f) {
+        //! @bug unk_4C8 and unk_4BC are used but not set (presumably intended to be feet positions like other actors)
+        func_800355B8(globalCtx, &this->unk_4C8);
+        func_800355B8(globalCtx, &this->unk_4BC);
+    }
+
+    if (SkelAnime_Update(&this->skelAnime) && (this->actor.bgCheckFlags & (1 | 2))) {
+        this->actor.world.rot.y = this->actor.shape.rot.y = this->actor.yawTowardsPlayer;
+        this->actor.shape.rot.x = 0;
+        this->actor.speedXZ = this->actor.velocity.y = 0.0f;
+        this->actor.world.pos.y = this->actor.floorHeight;
+
+        if (!Actor_OtherIsTargeted(globalCtx, &this->actor)) {
+            EnWf_SetupFirstSlash(this);
+        } else {
+            EnWf_SetupWait(this);
+        }
+    }
+}
+
+void EnWf_SetupReactToPlayer(EnWf* this) {
+    f32 lastFrame = Animation_GetLastFrame(&gWolfosBlockingAnim);
+
+    if (this->slashStatus != 0) {
+        this->slashStatus = -1;
+    }
+
+    this->actor.speedXZ = 0.0f;
+    this->action = WOLFOS_ACTION_REACT_TO_PLAYER;
+    this->actionTimer = 10;
+
+    Animation_Change(&this->skelAnime, &gWolfosBlockingAnim, 0.0f, 0.0f, lastFrame, ANIMMODE_ONCE_INTERP, -4.0f);
+    EnWf_SetupAction(this, EnWf_ReactToPlayer);
+}
+
+void EnWf_ReactToPlayer(EnWf* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s32 pad;
+
+    if (this->actionTimer != 0) {
+        this->actionTimer--;
+    } else {
+        this->skelAnime.playSpeed = 1.0f;
+    }
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        s16 yawDiff = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+
+        if ((!(ABS(yawDiff) > 0x4000)) && (this->actor.xzDistToPlayer < 60.0f) &&
+            (ABS(this->actor.yDistToPlayer) < 50.0f)) {
+            if (func_800354B4(globalCtx, &this->actor, 100.0f, 10000, 0x4000, this->actor.shape.rot.y)) {
+                if (player->swordAnimation == 0x11) {
+                    EnWf_SetupReactToPlayer(this);
+                } else {
+                    if ((globalCtx->gameplayFrames % 2) != 0) {
+                        EnWf_SetupReactToPlayer(this);
+                    } else {
+                        EnWf_SetupBackflip(this);
+                    }
+                }
```

Comment:
I think merging into an else if would be better, not sure

Reply:
it seems it might have been changed already, but i personally think it makes more sense as it was. the else makes it clear to me that the block below is relevant for any sword animation that is not 11. and the frames mod check is an either/or situation that belongs on its own i feel